digitizer/read_isotherm.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+"""Functions to read from existing JSON files"""
+import json
+
+from .config import QUANTITIES, find_by_key
+from .adsorbates import AdsorbateWithControls
+
+
+def lookup_species_name(species, species_type):
+    """Look up adsorbent or adsorbate species from a hash or name"""
+    if species_type == 'adsorbates':
+        key_type = 'InChIKey'
+    elif species_type == 'adsorbents':
+        key_type = 'hashkey'
+    else:
+        raise AttributeError(
+            'Species type {} not understood'.format(species_type))
+    if key_type in species.keys():
+        # Look up by hash first
+        output = find_by_key(species[key_type], key_type,
+                             QUANTITIES[species_type]['json'])
+    elif 'name' in species.keys():
+        # Fall back on name
+        output = species
+    else:
+        raise ValueError('Adsorbate or Adsorbent metadata in JSON incomplete')
+    return output['name']
+
+
+def read_isotherm_json(form):  # pylint: disable=unused-argument, disable=too-many-branches, disable=too-many-statements
+    # At present, only for single-component isotherms
```

Comment:
Is this still valid? I see some multicomponent code below...

Reply:
no, I neglected to remove it... fixed momentarily

Reply:
done